src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1592,11 +1592,7 @@ trait Namers extends MethodSynthesis {
       import SymValidateErrors._
       def fail(kind: SymValidateErrors.Value) = SymbolValidationError(sym, kind)
 
-      def checkWithDeferred(flag: Int) {
-        if (sym hasFlag flag)
-          AbstractMemberWithModiferError(sym, flag)
-      }
-      def checkNoConflict(flag1: Int, flag2: Int) {
+      def checkNoConflict(flag1: Int, flag2: Int) = {
```

Comment:
Apparently `-Ywarn-infer-any` doesn't snag on `if (cond) v` which is `if (cond) v else ()`. You must get Unit or AnyVal or Any.